src/System.Runtime.Extensions/tests/System/Environment.MachineName.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace System.Tests
+{
+    public class Environment_MachineName
+    {
+        // [Fact] Skip test for the mscolib changes to reach corefx
+        public void TestMachineNameProperty()
+        {
+            string computerName = GetComputerName();
+            Assert.Equal(computerName, Environment.MachineName);
+        }
+
+        [DllImport("kernel32.dll", EntryPoint = "GetComputerNameW")]
+        private unsafe static extern int GetComputerName(char* nameBuffer, ref int bufferSize);
+
+        [DllImport("libc")]
+        private unsafe static extern int gethostname(char* name, int len);
+
+        private unsafe static string GetComputerName()
+        {
+            const int MaxMachineNameLength = 256;
```

Comment:
The goal of the test here is not to test the functional correctness of the underlying OS APIs, but to test the .NET wrapper around it does not have bugs, or unexpected exceptions not being handled, etc. Additional check can be done, if ```InvalidOperationException``` is thrown, the test here is getting empty string from ```GetComputerName()```. 

Moreover this test is more robust than depending on envrionment variable which can be changed during program execution, and cause irrelevant failures. For ex, this code returns,

```c#
Console.WriteLine("{0} is from Environment.MachineName", Environment.MachineName);
Console.WriteLine("{0} is from Environment.GetEnvironmentVariable",Environment.GetEnvironmentVariable("COMPUTERNAME"));
Environment.SetEnvironmentVariable("COMPUTERNAME", "myprogramvalue");
Console.WriteLine("{0} is from Environment.MachineName", Environment.MachineName);
Console.WriteLine("{0} is from Environment.GetEnvironmentVariable", Environment.GetEnvironmentVariable("COMPUTERNAME"));
```

returns, 

```
LASEKAR1 is from Environment.MachineName
LASEKAR1 is from Environment.GetEnvironmentVariable
LASEKAR1 is from Environment.MachineName
myprogramvalue is from Environment.GetEnvironmentVariable
```